pgscatalog.core/src/pgscatalog/core/lib/models.py

```diff
@@ -0,0 +1,885 @@
+    )
+    other_allele: Optional[Allele] = Field(
+        default=None,
+        title="Other allele(s)",
+        description="The other allele(s) at the loci. Note: this does not necessarily need to correspond to the reference allele.",
+    )
+    locus_name: Optional[str] = Field(
+        default=None,
+        title="Locus Name",
+        description="This is kept in for loci where the variant may be referenced by the gene (APOE e4). It is also common (usually in smaller PGS) to see the variants named according to the genes they impact.",
+    )
+    is_haplotype: Optional[bool] = Field(
+        default=False,
+        title="FLAG: Haplotype",
+        description="This is a TRUE/FALSE variable that flags whether the effect allele is a haplotype/diplotype rather than a single SNP. Constituent SNPs in the haplotype are semi-colon separated.",
+    )
+    is_diplotype: Optional[bool] = Field(
+        default=False,
+        title="FLAG: Diplotype",
+        description="This is a TRUE/FALSE variable that flags whether the effect allele is a haplotype/diplotype rather than a single SNP. Constituent SNPs in the haplotype are semi-colon separated.",
+    )
+    imputation_method: Optional[str] = Field(
+        default=None,
+        title="Imputation Method",
+        description="This described whether the variant was specifically called with a specific imputation or variant calling method. This is mostly kept to describe HLA-genotyping methods (e.g. flag SNP2HLA, HLA*IMP) that gives alleles that are not referenced by genomic position.",
+    )
+    variant_description: Optional[str] = Field(
+        default=None,
+        title="Variant Description",
+        description="This field describes any extra information about the variant (e.g. how it is genotyped or scored) that cannot be captured by the other fields.",
+    )
+    inclusion_criteria: Optional[str] = Field(
+        default=None,
+        title="Score Inclusion Criteria",
+        description="Explanation of when this variant gets included into the PGS (e.g. if it depends on the results from other variants).",
+    )
+
+    # weight information
+    # all effect weight fields are handled as strings internally to faithfully reproduce author-uploaded scores (i.e. avoid any floating point errors)
+    effect_weight: Optional[str] = Field(
+        default=None,
+        title="Variant Weight",
+        description="Value of the effect that is multiplied by the dosage of the effect allele (effect_allele) when calculating the score. Additional information on how the effect_weight was derived is in the weight_type field of the header, and score development method in the metadata downloads.",
+        coerce_numbers_to_str=True,
```

Comment:
Will this cause problems with precision?

Reply:
(Summary from our slack chat)

This affects instantiating the class. Without string coercion this will fail:

```python
CatalogScoreVariant(**{"effect_weight": 0.1, ...})
```

So you'd have to do:

```python
CatalogScoreVariant(**{"effect_weight": "0.1", ...})
```

Which is mildly annoying. Internally we always load weights as strings. It's up to users of this model to make sure they use strings to instantiate the class if they're worried about precision. I added a better explanation here:

https://github.com/PGScatalog/pygscatalog/blob/b379292cccbb0bff3f23590fe9c35e3e4c5e968a/pgscatalog.core/src/pgscatalog/core/lib/models.py#L102-L139